docs/tutorials/ipfs-gallery/index.md

```diff
@@ -0,0 +1,586 @@
+        <div><img src="https://picsum.photos/1920/1200?rand32" /></div>
+        <div><img src="https://picsum.photos/1920/1080?rand33" /></div>
+      </div>
+      <div class="row">
+        <div><img src="https://picsum.photos/1920/1080?rand41" /></div>
+      </div>
+  </div>
+```
+
+Notice how things still jump around while loading? We can fix this by specifying the width and the height in the `<img>` elements:
+
+```html
+  <div class="gallery">
+      <div class="row">
+        <div><img src="https://picsum.photos/1920/1080?rand11" width="1920" height="1080"/></div>
+        <div><img src="https://picsum.photos/1040/1080?rand12" width="1040" height="1080"/></div>
+        <div><img src="https://picsum.photos/1080/1920?rand13" width="1080" height="1920"/></div>
+        <div><img src="https://picsum.photos/225/400?rand14" width="225" height="400"/></div>
+      </div>
+      <div class="row">  
+        <div><img src="https://picsum.photos/1080/1920?rand21" width="1080" height="1920"/></div>
+        <div><img src="https://picsum.photos/1080/1920?rand22" width="1080" height="1920"/></div>
+        <div><img src="https://picsum.photos/1920/1080?rand23" width="1920" height="1080"/></div>
+        <div><img src="https://picsum.photos/1920/1080?rand24" width="1920" height="1080"/></div>
+      </div>
+      <div class="row">
+        <div><img src="https://picsum.photos/1920/1080?rand31" width="1920" height="1080"/></div>
+        <div><img src="https://picsum.photos/1920/1200?rand32" width="1920" height="1200"/></div>
+        <div><img src="https://picsum.photos/1920/1080?rand33" width="1920" height="1080"/></div>
+      </div>
+      <div class="row">
+        <div><img src="https://picsum.photos/1920/1080?rand41" width="1920" height="1080"/></div>
+      </div>
+  </div>
+```
+
+To make the layout work on mobile, we will use a [CSS media query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_media_queries/Using_media_queries) that will change the layout so that images are each the full with of the screen rather than forming a complex grid. By adding
+
+```
+      @media (width < 678px) {
+          .gallery > .row {
+              aspect-ratio: unset !important;
+              flex-direction: column;
+          }
+          .gallery .row img {
+              height: auto;
+              width: 100%;
+          }
+      }
+```
+
+And let's get rid of the bit of margin around the edge, add the following CSS rule in the style element:
+
+```css
+    body { 
+        margin: 0; 
+    }
+```
+
+The last thing we'll add to the gallery is the ability to view individual images. Add a new child element to the gallery div:
+
+```html
+<div class="lightBox hidden"><img src="https://picsum.photos/1920/1080" /></div>
+```
+
+And then let's update the styling for this to display as an overlay. We will use absolute positioning to make it cover the whole screen
+
+```css
+      .hidden {
```

Comment:
Instead of a class, could you get into using the `[aria-hidden]` attribute?

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden

Reply:
I'm not sure if I understand correctly?

As per the docs linked  `aria-hidden` won't visually hide elements:

> An element's hidden status is based on whether it is rendered. Rendering is usually controlled by CSS. For example, an element whose display property is set to none via CSS is not rendered. An element is considered hidden if it, or any of its ancestors are not rendered or have their aria-hidden attribute value set to true. Note that an element and its children that has aria-hidden="true" declared on it will still be visible, unless also hidden by CSS.

And I take it that I shouldn't be adding `aria-hidden` to elements I'm hiding using `display: none`.

> aria-hidden="true" should not be added when:
> -    The HTML [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute is present
> -    The element or the element's ancestor is hidden with [display: none](https://developer.mozilla.org/en-US/docs/Web/CSS/display)
> -    The element or the element's ancestor is hidden with [visibility: hidden](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility)

Reply:
Oh, fair. disregard this then. :+1: